### ISSUE TITLE
PyMuPDF==1.24.0 will hanging when using page.get_text("text")

### ISSUE THREAD

**Issue by xiaominghero** [pymupdf/PyMuPDF#3357]
### Description of the bug

os: linux Ubuntu 22.04 LTS
python 3.10.2

When I upload a PDF file, the program hangs for several hours without exiting When using get_text() method.

![image](https://github.com/pymupdf/PyMuPDF/assets/20225512/2a945b3c-df79-4bd9-a4f2-222e36530a96)

### How to reproduce the bug

```shell
>>> import fitz as pymupdf
>>> pdf_path = '/data/dataset/book/pdf/bad/e4a0626f933941c6db3257f5cea4f3e5.pdf'
>>> def parse_test(pdf_path) -> str:
...     pymu_doc = pymupdf.open(pdf_path, filetype="pdf")
...     contents = []
...     try:
...         if not pymu_doc:
...             return contents
...         for _, page in enumerate(pymu_doc):
...             content = page.get_text("text")
...             contents.append(content.replace('\n', ' '))
...     except Exception:
...         contents = []
...     return '\n'.join(contents)
...
>>> a = parse_test(pdf_path)
``` 

[e4a0626f933941c6db3257f5cea4f3e5.pdf](https://github.com/pymupdf/PyMuPDF/files/14896522/e4a0626f933941c6db3257f5cea4f3e5.pdf)

### PyMuPDF version

1.24.0

### Operating system

Linux

### Python version

3.10

**Comment by JorjMcKie** [comment]
This is base library problem occurring on the first page - the second page works.
I will open a bug in MuPDF's issue system.

**Comment by JorjMcKie** [comment]
When providing code snippets, please use properly indented code blocks using slash commands 
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/502418e9-d3ca-4398-a612-a341bafee45d)

**Comment by JorjMcKie** [comment]
Here is the reference to the MuPDF bug: https://bugs.ghostscript.com/show_bug.cgi?id=707721

**Comment by JorjMcKie** [comment]
> Interestingly enough, it was triggered on a .pdf with the same first page as @xiaominghero shared (attached).

It is this one page that causes the problem - not any other.

**Comment by JorjMcKie** [comment]
This has been fixed in v1.24.2.

**Comment by henriklaurentz** [comment]
Hi @JorjMcKie ,

I still run into the same issues on `1.24.2`  while testing on the file (the first page) that @jan-benisek submitted.

That is, 

```python
import fitz

fitz.open("I_break_things.pdf")[0].get_text()
```
hangs, while working fine on `1.23.14`.

**Comment by JorjMcKie** [comment]
Sorry - my bad. The required / developed MuPDF fix is **_not_** contained in this version yet.
I am re-opening the issue.

**Comment by hnkelau** [comment]
No worries, and thanks a lot for all your work.

**Comment by henriklaurentz** [comment]
No worries, and thanks a lot for all your work.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.3.
